### PR TITLE
Default to `go1.17.11`

### PR DESCRIPTION
**Pull request by timvaillancourt**
### Description

This PR defaults the build to go1.17.11

This resolves https://github.com/github/gh-ost/issues/1128

> In case this PR introduced Go code changes:

- [x] contributed code is using same conventions as original code
- [x] `script/cibuild` returns with no formatting errors, build errors or unit test errors.

### CODE REVIEW

**Review by Nholam1990 — COMMENTED** [review]
vendor/modules.txt

File: vendor/github.com/go-sql-driver/mysql/go.mod [hunk 1]
```diff
@@ -1,3 +0,0 @@
-module github.com/go-sql-driver/mysql
```
Comment (FromAshWeRise) [inline vendor/github.com/go-sql-driver/mysql/go.mod:1]: @timvaillancourt   what am I looking at here?

File: vendor/github.com/pingcap/errors/go.sum [hunk 2]
```diff
@@ -1,66 +0,0 @@
-github.com/BurntSushi/toml v0.3.1 h1:WXkYYl6Yr3qBf1K79EBnL4mak0OimBfB0XUf9Vl28OQ=
-github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=
-github.com/davecgh/go-spew v1.1.0/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
-github.com/davecgh/go-spew v1.1.1 h1:vj9j/u1bqnvCEfJOwUhtlOARqs3+rkHYY13jYWTU97c=
-github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
-github.com/google/renameio v0.1.0/go.mod h1:KWCgfxg9yswjAJkECMjeO8J8rahYeXnNhOm40UhjYkI=
```
Comment (FromAshWeRise) [inline vendor/github.com/pingcap/errors/go.sum:6]: did y'all not like the name I gave it?

File: vendor/github.com/go-sql-driver/mysql/go.mod (hunk 1 above)
Comment (timvaillancourt) [reply vendor/github.com/go-sql-driver/mysql/go.mod:1]: @FromAshWeRise the files in `vendor/` are the result of `go mod vendor` + `git add vendor --all`, so `go mod` wanted this somehow

**Pull request merged**